kubecontrolplane/v1/types.go

```diff
@@ -62,6 +62,13 @@ type KubeAPIServerConfig struct {
 
 	// TODO this needs to be removed.
 	APIServerArguments map[string]Arguments `json:"apiServerArguments"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny most authorization requests of kubelets that are older
+	// than the specified version, only allowing the kubelet to get and update its node object, and perform
+	// subjectaccessreviews.
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	MinimumKubeletVersion string `json:"minimumKubeletVersion,omitempty"`
```

Comment:
Once we agree on godoc and validations on the other version of this field, please make sure to copy it here

Reply:
I decided to skip the validations because it didn't look like the other fields in this file were being validated. should I add them?

Reply:
Please. Over time we want to improve validation and even if validations aren't present for existing fields, we shouldn't use that as a pattern. Future tooling will enforce a lot of validation and leaving this now, will produce exceptions and toil in the future

Reply:
You know that, and I now know that, but will future people who decide to copy this API? I realise it does nothing, but I'd rather we have it here for completeness, thanks for updating